### PR TITLE
Add and update output coordinate handling

### PR DESCRIPTION
**Pull request by dallasfoster**
# Earth2Studio Pull Request

## Description
This PR changes the behavior of output coordinate handling in `px` and `dx` models, as well as in `statistics`. The intent of this PR is to make sure that each method can specify what the output coordinate system is for a given input coordinate system. 

For all models the `output_coords` utility function should provide the resulting coordinates after using the associated `__call__` method for each model. For prognostic models this method, similar to the `__call__` method, may not necessarily be run autoregressively and there is not equivalent iterator method for computing output coordinates.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/earth2studio/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/earth2studio/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/earth2studio/issues) is linked to this pull request.

### CODE REVIEW

File: earth2studio/models/px/fcn.py [hunk 1]
```diff
@@ -176,14 +196,17 @@ def __call__(
         tuple[torch.Tensor, CoordSystem]
             Output tensor and coordinate system 6 hours in the future
         """
+
         for i, (key, value) in enumerate(self.input_coords.items()):
```
Comment (NickGeneva) [inline earth2studio/models/px/fcn.py]: Can this get moved up into `ouput_coords`?

File: earth2studio/models/px/fengwu.py [hunk 2]
```diff
@@ -349,7 +368,7 @@ def __call__(
                 handshake_dim(coords, key, i)
```
Comment (NickGeneva) [inline earth2studio/models/px/fengwu.py]: Move up into output coords
If you do this then `self.output_coords(coords)` should be called right away prior to forward pass to trigger coord errors before model attempts to execute.

I.e.
```
output_coords = self.output_coords(coords)
return self._forward(x, self.ort), output_coords

File: earth2studio/models/px/fuxi.py [hunk 3]
```diff
@@ -449,20 +473,23 @@ def _default_generator(
...
-            # Use output as next input
-            x = out
-            coords = out_coords.copy()
-
+            # Yield current output
             output = out[:, :, 1:]
-            out_coords["lead_time"] = out_coords["lead_time"][1:]
             yield output, out_coords.copy()
 
+            # Use output as next input
```
Comment (NickGeneva) [inline earth2studio/models/px/fuxi.py:426]: double comment

File: earth2studio/models/dx/climatenet.py [hunk 4]
```diff
@@ -79,23 +79,60 @@ def __init__(
...
+            return OrderedDict(
+                {
+                    "batch": np.empty(0),
+                    "variable": np.array(OUT_VARIABLES),
+                    "lat": np.linspace(90, -90, 721, endpoint=True),
+                    "lon": np.linspace(0, 360, 1440, endpoint=False),
+                }
+            )
+
+        handshake_dim(input_coords, "lon", 3)
```
Comment (NickGeneva) [inline earth2studio/models/dx/climatenet.py:116]: having the handshake methods in the output coords seems like a good call, should this be the expectation for all models?

File: earth2studio/models/px/fengwu.py [hunk 5]
```diff
@@ -163,15 +163,40 @@ def __init__(
...
-            "batch": np.empty(0),
-            "lead_time": np.array([np.timedelta64(6, "h")]),
-            "variable": np.array(VARIABLES),
-            "lat": np.linspace(90, -90, 721, endpoint=True),
-            "lon": np.linspace(0, 360, 1440, endpoint=False),
-        }
-    )
+    @batch_coords()
+    def output_coords(self, input_coords: CoordSystem | None = None) -> CoordSystem:
+        """Ouput coordinate system of the prognostic model
```
Comment (NickGeneva) [inline earth2studio/models/px/fengwu.py]: "Output"

Comment (NickGeneva) [reply earth2studio/models/px/fengwu.py]: ctrl-f ouput, seems in multiple files

File: earth2studio/models/px/fcn.py (hunk 1 above)
Comment (dallasfoster) [reply earth2studio/models/px/fcn.py]: Done.

File: earth2studio/models/px/fuxi.py (hunk 3 above)
Comment (dallasfoster) [reply earth2studio/models/px/fuxi.py:426]: removed.

File: earth2studio/models/dx/climatenet.py (hunk 4 above)
Comment (dallasfoster) [reply earth2studio/models/dx/climatenet.py:116]: Yes, I am moving all of the handshakes to the `output_coords` method.

File: earth2studio/models/px/fengwu.py (hunk 5 above)
Comment (dallasfoster) [reply earth2studio/models/px/fengwu.py]: Thanks, fixed.

File: earth2studio/models/px/fengwu.py (hunk 2 above)
Comment (dallasfoster) [reply earth2studio/models/px/fengwu.py]: Done

**Review by NickGeneva — APPROVED** [review]

**Pull request merged**